packages/babel-plugin-proposal-class-properties/test/fixtures/compile-to-class/preserve-comments/output.js

```diff
@@ -3,8 +3,8 @@ class C {
     /* before a */
     babelHelpers.defineProperty(this, "a", void 0);
   }
+  /* after a */
+  /* before b */
   /* after b */
 }
-/* after a */
-/* before b */
 babelHelpers.defineProperty(C, "b", void 0);
```

Comment:
IMO in this case the old behaviour is more preferable, `/* before b */` is more likely to be the JSDoc comment of class property `b`.

Reply:
What do you think should be the priority? When we remove node B from ABC, we first try to attach B's trailing comments as leading comments of C, and then fallback to trailing comments of A?

Reply:
Ok actually, the priority I implemented is already like that.

What's happening here is that when we have this code:
```js
class C {
  constructor() {}
  /* before b */
  static b;
  /* after b */
}
```
we call `path.remove()` on `static b`, and thus its comments are attached to `constructor() {}` which is the only available sibling node.

I'm confused by the previous output, because `after b` was attached to the constructor but `before b` wasn't :thinking:

Reply:
I pushed a solution, I'm not sure I like it. The behavior I was confused about was because of the helpers for class fields to inherit comments that we recently introduced.